src/main/java/uk/gov/hmcts/ccd/domain/service/casedataaccesscontrol/DefaultCaseDataAccessControl.java

```diff
@@ -81,7 +81,10 @@ public DefaultCaseDataAccessControl(RoleAssignmentService roleAssignmentService,
     public Set<AccessProfile> generateAccessProfilesByCaseReference(String caseReference) {
         Optional<CaseDetails> caseDetails =  caseDetailsRepository.findByReference(caseReference);
         if (caseDetails.isEmpty()) {
-            return Sets.newHashSet();
+            caseDetails = caseDetailsRepository.findById(null,Long.parseLong(caseReference));
```

Comment:
Maybe we should have a comment for this line to explain why we're doing this - I assume to cover scenarios where a case ID is provided instead of the reference?